<launch>
    <!-- ARUCO -->
    <include file="$(find human_robot_collaboration)/launch/baxter_aruco.launch">
        <arg name="arm" value= "left" />
    </include>

    <!-- BAXTER DISPLAY -->
    <include file="$(find human_robot_collaboration)/launch/baxter_display.launch">
        <arg name="onscreen_duration" value = "16.0" />
    </include>

    <!-- ROSBRIDGE -->
    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" />

    <!-- HSV DETECTION -->
    <node name="image_view_hsv" pkg="image_view" type="image_view" respawn="false" output="log">
        <remap from="image" to="/hsv_detector/image_result"/>
        <param name="autosize" value="true" />
    </node>

    <!-- <node name="image_view_hsv2" pkg="image_view" type="image_view" respawn="false" output="log">
        <remap from="image" to="/hsv_detector/image_result_thres"/>
        <param name="autosize" value="true" />
    </node> -->

    <param name="hsv_detector/reference_frame" value="/base"/>
    <param name="hsv_detector/camera_frame"    value="/right_hand_camera"/>
    <param name="hsv_detector/area_threshold"  value="250"/>

    <node pkg="human_robot_collaboration" type="hsv_detector" name="hsv_detector" output="screen" respawn="true">
        <remap from="hsv_detector/image" to="/cameras/right_hand_camera/image"/>
        <remap from="hsv_detector/camera_info" to="/cameras/right_hand_camera/camera_info"/>
    </node>

    <!-- Objects database for the left arm -->
    <rosparam param = "action_provider/objects_left">
        "table_top"   : 200
        "leg_1"       : 150
        "leg_2"       : 151
        "leg_3"       : 152
        "leg_4"       : 153
    </rosparam>

    <!-- Objects database for the right arm -->
    <rosparam param = "action_provider/objects_right">
        "screwdriver"  : 1
        "screws_box"   : 2
        "brackets_box" : 3
    </rosparam>

    <!-- Objects database for the hsv cartesian detector -->
    <rosparam param = "hsv_detector/objects_db">
        "screwdriver" : { id: 1, size: [0.099, 0.030], HSV: { H: [160,  10], S: [ 70, 166], V: [10,  66] } }
         <!-- This should be the blue box -->
        "screws_box"  : { id: 2, size: [0.111, 0.111], HSV: { H: [ 80, 130], S: [120, 216], V: [70, 146] } }
         <!-- This should be the green box -->
        "brackets_box": { id: 3, size: [0.111, 0.111], HSV: { H: [ 30, 100], S: [ 80, 246], V: [20,  96] } }
    </rosparam>

    <!-- CONTROL -->
    <!-- If the system should use the robot or not (useful for debugging) -->
    <arg name="useRobot"            default="true"/>
    <!-- If the system should use the robot or not (useful for debugging) -->
    <param name="action_provider/use_robot"             value="$(arg useRobot)"/>
    <rosparam param="/print_level">0</rosparam>

    <node pkg="human_robot_collaboration" type="modular_action_provider" name="modular_action_provider" output="screen">
        <remap from="/markers/left"  to="/baxter_aruco_left/markers"/>
        <remap from="/markers/right" to="/hsv_detector/objects"/>
    </node>

    <!-- VISUALIZATION & MISC -->

    <include file="$(find svox_tts)/launch/svox_tts.launch" />

    <!-- Send recognize speech to baxter screen -->
    <remap from="/speech_to_text/text" to="/svox_tts/speech_output"/>
    <include file="$(find ros_speech2text)/launch/ros_speech2text.launch"/>

</launch>

